test/integration_test/execute_rule/transfer_with_global_constraint.js

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** @dev  This is the integration test with global constraint.
+ *        Global constraint restricts sum of all the tokens to be transferred
+ *        is not more than the set limit.
+ *        In the test, we have set the limit as 200 tokens in LimitTransferGlobalConstraint
+ *        contract.But,we try to transfer 250 tokens in a transaction.
+ *
+ *        Following steps are performed in the test :-
+ *
+ *        - EIP20TokenMock contract is deployed.
+ *        - Organization contract is deployed and worker is set.
+ *        - TokenRules contract is deployed.
+ *         TransferRule contract is deployed and it is registered in TokenRules.
+ *        - TokenHolder contract is deployed by providing the wallets and
+ *           required confirmations.
+ *        - LimitTransferGlobalConstraint contract is deployed. It is registered
+ *          in the TokenRules as global constraint. The limit per transaction is
+ *          200 tokens.
+ *        - Using EIP20TokenMock's setBalance method,tokens are provided to TH.
+ *        - We generate executable data for TransferRule contract's transferFrom
+ *           method.
+ *        - Relayer calls executeRule method of tokenholder contract.
+ *           After it's execution below verifications are done:
+ *            - RuleExecuted event.
+ *            - tokenholder balance.
+ *            - 'to' address balance.
+ */
+const EthUtils = require('ethereumjs-util'),
+  utils = require('../../test_lib/utils'),
+  AccountsProvider = utils.AccountProvider,
+  ExecuteRuleUtils = require('./utils'),
+  BN = require('bn.js'),
+  LimitTransferGlobalConstraint = artifacts.require('LimitTransferGlobalConstraint'),
+  { Event } = require('../../test_lib/event_decoder');
+
+contract('TokenHolder::executeRule', async (accounts) => {
+
+  let accountProvider,
+    tokenHolder,
+    wallet1,
+    eip20TokenMock,
+    transferRule,
+    tokenRules,
+    ephemeralPrivateKey1,
+    ephemeralKeyAddress1,
+    keyData,
+    totalBalance = 500,
+    transferLimit = 200,
+    limitTransferGlobalConstraint,
+    worker;
+
+  describe('ExecuteRule integration test', async () => {
+
+    it('Should fail while transferring BTs more than global constraint on max tokens to be transferred', async () => {
+
+      accountProvider = new AccountsProvider(accounts);
+
+      ( {
```

Comment:
no need of "(" and ")".
You con simply use like below for consistency:
const  {tokenHolder,
        wallet1,
        eip20TokenMock,
        transferRule,
        tokenRules,
        worker,
      } = await ExecuteRuleUtils.setup(accountProvider);

Reply:
If we use above suggested way, we will have to deploy all the contracts in every **it** .